### PR TITLE
remote: fixed Repo.push() to default to 1 job

### PR DESCRIPTION
**Pull request by Suor**

### CODE REVIEW

File: dvc/repo/push.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 def push(
     self,
     target=None,
-    jobs=1,
+    jobs=None,
```
Comment (efiop) [inline dvc/repo/push.py]: What about repo/pull.py and others?

Comment (Suor) [reply dvc/repo/push.py]: Didn't notice, but they also has 1 as default. Looks like this was done on purpose. I wonder, which one and whether that still stands.

**Comment by Suor** [comment]
Dropped `jobs=1` defaults for all methods I found. Looks like we had several defaults this 1 and the one in `Remote*.JOBS`, now we only have the latter.

**Comment by Suor** [comment]
Have no idea why this fails with pyarrow not installed on Win-Py37. @efiop?

**Comment by efiop** [comment]
@Suor Just a bug in new pyarrow. https://issues.apache.org/jira/browse/ARROW-6015?filter=-6 I've frozen the required version to 0.14.0 for now, which should do the trick.

**Pull request merged**